src/ADAL.PCL/Http/AdalHttpClient.cs

```diff
@@ -105,12 +111,29 @@ public AdalHttpClient(string uri, CallState callState)
                         serviceEx = new AdalServiceException(tokenResponse.Error, tokenResponse.ErrorDescription,
                             errorCodes, ex);
 
+                        if ((int)ex.WebResponse.StatusCode == 400 && tokenResponse.Error == interactionRequired)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 control flow statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS134)